client/matching/loadbalancer.go

```diff
@@ -70,6 +71,28 @@ type (
 		nWritePartitions    dynamicconfig.IntPropertyFnWithTaskQueueInfoFilters
 		forceReadPartition  dynamicconfig.IntPropertyFn
 		forceWritePartition dynamicconfig.IntPropertyFn
+
+		lock         sync.Mutex
```

Comment:
How often will we be writing to this map? Are we expecting a high write load? If not, we may wish to prefer an RWMutex (and we could just upgrade the lock if our entry is missing)

Reply:
not much, whenever a TQ is seen for the first time in this process.

Reply:
good point. please review the new implementation.